### DISCUSSION TITLE
How to get pathname in App Page?

### DISCUSSION THREAD

**Discussion by guotie** [vercel/next.js#57312] [App Router]
Hi, guys, there is a problems:

how to get pathname in App page;

Layout and Page is server components, I think the server components should be able to access the hole `request`, but now I cannot even get the pathname, It's weired

**Comment by mass2527** [comment]
Unfortunately, there's no recommended way to access pathname from server components.
https://github.com/vercel/next.js/issues/43704#issuecomment-1566347726
